kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyReconciler.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.kubernetes.operator;
+
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.ControllerConfiguration;
+import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
+import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
+import io.javaoperatorsdk.operator.api.reconciler.dependent.Dependent;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyBuilder;
+
+@ControllerConfiguration(dependents = {
+        @Dependent(type = ProxyConfigSecret.class),
+        @Dependent(type = ProxyDeployment.class),
+        @Dependent(type = ProxyService.class)
+})
+public class ProxyReconciler implements Reconciler<KafkaProxy> {
+
+    public UpdateControl<KafkaProxy> reconcile(KafkaProxy primary,
+                                               Context<KafkaProxy> context) {
+        return UpdateControl.patchStatus(
```

Comment:
Answering at least partly my previous comment, the logic I was expecting to see here is in `*Depenant#desired`. 

So when in the life cycle is `reconcile` called? Is it prior to the dependants being processed or after? If its prior should we not be doing something to invoke `desired` on each dependant?